server/apps/immich/src/api-v1/share/shared-link.repository.ts

```diff
@@ -0,0 +1,63 @@
+import { SharedLinkEntity } from '@app/database/entities/shared-link.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { Logger } from '@nestjs/common';
+
+export interface ISharedLinkRepository {
+  get(userId: string): Promise<SharedLinkEntity[]>;
+  getbyId(id: string): Promise<SharedLinkEntity | null>;
+  getByKey(key: string): Promise<SharedLinkEntity | null>;
+  create(payload: SharedLinkEntity): Promise<SharedLinkEntity>;
+  remove(entity: SharedLinkEntity): Promise<SharedLinkEntity>;
+  save(entity: SharedLinkEntity): Promise<SharedLinkEntity>;
+}
+
+export const ISharedLinkRepository = 'ISharedLinkRepository';
+
+export class SharedLinkRepository implements ISharedLinkRepository {
+  readonly logger = new Logger(SharedLinkRepository.name);
+  constructor(
+    @InjectRepository(SharedLinkEntity)
+    private readonly sharedLinkRepository: Repository<SharedLinkEntity>,
+  ) {}
+
+  async get(userId: string): Promise<SharedLinkEntity[]> {
+    return await this.sharedLinkRepository.find({
+      where: {
+        userId: userId,
+      },
+      relations: ['assets', 'album'],
+    });
+  }
+
+  async create(payload: SharedLinkEntity): Promise<SharedLinkEntity> {
+    return await this.sharedLinkRepository.save(payload);
+  }
+
+  async getbyId(id: string): Promise<SharedLinkEntity | null> {
+    return await this.sharedLinkRepository.findOne({
+      where: {
+        id: Buffer.from(id, 'hex'),
```

Comment:
I think these types of conversions should no happen in the repo. Buffer's should pretty much be used everywhere, with the exception of the controller/guard, which should convert from buffer to string using a method in the share core.